src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -378,7 +378,9 @@ void primitive_inst::update_shape() {
                 break;
             }
         }
-        if (!subgraph_input_changed) {
+        if (subgraph_input_changed) {
+            set_flag(ExecutionFlags::SHAPE_CHANGED);
```

Comment:
I guess it should have been already turned on from line 359. why isn't it set previously?